test/pkg/environment/aws/expectations.go

```diff
@@ -563,3 +571,105 @@ func ExpectCapacityReservationsCanceled(ctx context.Context, ec2api *ec2.Client,
 		Expect(err).ToNot(HaveOccurred())
 	}
 }
+
+func (env *Environment) EventuallyExpectRoleCreated(roleName string) {
```

Comment:
nit: I'd call this `EventuallyExpectNodeRoleCreated` since the trust policy and attached policies are really only for a node role.